pkg/providers/aws/provider_postgres.go

```diff
@@ -214,6 +219,87 @@ func (p *AWSPostgresProvider) createRDSInstance(ctx context.Context, cr *v1alpha
 	}}, croType.StatusMessage(fmt.Sprintf("creation successful, aws rds status is %s", *foundInstance.DBInstanceStatus)), nil
 }
 
+// Tags RDS resources
+func (p *AWSPostgresProvider) TagRDSPostgres(ctx context.Context, cr *v1alpha1.Postgres, foundInstance *rds.DBInstance) (croType.StatusMessage, error) {
+	logrus.Infof("Adding Tags to RDS instance %s", *foundInstance.DBInstanceIdentifier)
+	// get the environment from the CR
+	defaultOrganizationTag, exists := os.LookupEnv("TAG_KEY_PREFIX")
+	if !exists {
+		defaultOrganizationTag = TagKeyPrefex
+	}
+
+	// new session but don't need credentials
+	// Get the region from AvailabilityZone
+	rdsRegion := *foundInstance.AvailabilityZone
+	rdsRegion = rdsRegion[:(len(rdsRegion))-1]
+	// get provider aws creds so the postgres instance can be deleted
+	rdsSvc := rds.New(session.Must(session.NewSession(&aws.Config{
```

Comment:
We need to define the credentials here and/or define the RDS/Elasticache sessions the same way as we do in existing parts of the code. Otherwise AWS attempts to use your on-machine credentials which won't exist in a pod.

There are existing examples of using these credentials in the pod. Happy to contribute a change/pair up if wanted.

This is the error I end up seeing in the CRs

```
  message: "Failed to add tags to RDS resources: Failed to add Tags to RDS instance: NoCredentialProviders: no valid providers in chain. Deprecated.\n\tFor verbose messaging see aws.Config.CredentialsChainVerboseErrors"
```

Reply:
:+1: